actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -482,7 +477,7 @@ def collection_without_template(view)
           template = (cache[path] ||= find_template(path, keys + [as, counter, iteration]))
           content = template.render(view, locals)
           partial_iteration.iterate!
-          content
+          build_rendered_template(content, nil, template)
```

Comment:
The nil in here reads a little confusing. How about making layout an optional kw arg? Then we can just omit nil here.